earthfile2llb/converter.go

```diff
@@ -1140,7 +1140,10 @@ func (c *Converter) Expose(ctx context.Context, ports []string) error {
 	}
 	c.nonSaveCommand()
 	for _, port := range ports {
-		c.mts.Final.MainImage.Config.ExposedPorts[port] = struct{}{}
+		if strings.Contains(port, ":") {
+			port = strings.SplitN(port, ":", 2)[1]
```

Comment:
why is the first value being ignored?

Reply:
since container will be exposed on 2nd port, we are getting only 2nd port.
for example, if value is 8080:8081, then 8081 gets set as container port

Reply:
what does the first value represent? It seems odd that we can safely ignore it. Can you add a comment pointing to some documentation which backs up your reasoning.

Reply:
first value represents host port. https://docs.docker.com/engine/reference/run/#expose-incoming-ports this provides more detail

Reply:
That's a link to the cli reference; if nothing exists specifically for the dockerfile reference, maybe you can link to some code showing that docker (or buildkit) ignores this value too.

Reply:
https://github.com/earthly/buildkit/blob/earthly-main/vendor/github.com/docker/go-connections/nat/nat.go#L176 this function ignores the first port and sets second port as container port.

Reply:
https://github.com/earthly/buildkit/blob/earthly-main/vendor/github.com/docker/go-connections/nat/nat.go#L176 this function is separating container and host port. docker uses container port to communicate to host. if host port is not given, docker allocates that automatically. hence when we provide <port1>:<port2> it assumes port1 is host port and port2 is container port

Reply:
2 containers I ran with shown dockerfile but ran with -P & -p options to expose all ports in the dockerfile or only particular ones, respectively. Note the exposed ports for both.
![Screenshot from 2022-08-09 19-15-27](https://user-images.githubusercontent.com/8165377/183712829-1411310c-a7df-4e3f-9a9f-5f204389e8d6.png)

 Contents of file /proc/sys/net/ipv4/ip_local_port_range on Ubuntu 20.04
![Screenshot from 2022-08-09 19-20-37](https://user-images.githubusercontent.com/8165377/183712853-97bdc699-83fc-4618-9909-d2396455a5c5.png)

Docker inspect output for the 2 containers:
![Screenshot from 2022-08-09 19-44-31](https://user-images.githubusercontent.com/8165377/183712920-e2a8f3c6-f0e5-4e7f-9742-6abd25c75eef.png)